### ISSUE TITLE
Games in different libraries are not linked correctly

### ISSUE THREAD

**Issue by Snogard** [Jannomag/shortix#7]
Steps to reproduce:
1. Add a second steam library and install a new game.
1. run shortix

the resulting symlinks will point to the default steam location regardless of where the actual game is installed

**Comment by Jannomag** [comment]
I'm quiet busy at the moment, as you can see at my late respond. Sorry for this. Whenever find enough time to fix things on Shortix I will do it, but please have patience with me. This feature seems complex to develop - at least for me. I'll do my best!

**Comment by Snogard** [comment]
don't worry and take your time, this is a foss project after all :)
Anyway you may take a look at `$HOME/.steam/steam/config/libraryfolders.vdf` to read where apps are installed and this [python library](https://pypi.org/project/vdf/) to read vdf files

**Comment by Jannomag** [comment]
Python is a new language for me, but I’ll try my best. Thanks for the tip

**Comment by Spiricore** [comment]
I ran into this same issue and was wondering if you had a way to resolve it? I have another drive (which happens to be my main game drive) I'd like shortix to symlink.

Thank you for this tool, it has made my life so much better on my steam deck!

**Comment by Jannomag** [comment]
Sorry for the late reply on the created issue, I still have it mind but the lack of time is terrible and I'm focused on other things.

I still have it in mind. For now my steam deck is broken and until I fixed I'm not able to work any further on this without proper testing.

Since I keep all my steam generated prefixes on my main drive I didn't think about this possibility.

**Comment by Snogard** [comment]
if i were to make a pull request, would you have time to look at it? i may do one in the coming week or two

**Comment by Jannomag** [comment]
> if i were to make a pull request, would you have time to look at it? i may do one in the coming week or two

Yes, if you can test it before creating it. I'll check your code and merge it

**Comment by Snogard** [comment]
i ended up making the bulk of the update today XD
I just need to confirm if the steam deck has pip installed or not. can you check @Spiricore ?

**Comment by Spiricore** [comment]
> i ended up making the bulk of the update today XD I just need to confirm if the steam deck has pip installed or not. can you check [@Spiricore](https://github.com/Spiricore) ?

It does not have pip installed by default, but python 3.10.8 is (if that is relevant).

**Comment by Snogard** [comment]
Alright, i think i have a stable version up on [my fork](https://github.com/Snogard/shortix/tree/improvements).

Since i don't have a SteamDeck i cannot test there but since i'm running Arch myself i'm very confident it should work, so @Spiricore if you want to give it a try and report back that would be appreciated :)

Just clone my repo, swith to the "improvements" branch and run "simulate_installer.sh"
If all goes well i'll make a merge request.

### What i've changed beside adding the feature:
- Moved config files to `$HOME/.config/Shortix`
- Moved run files to `$HOME/.cache/Shortix`
- Removed `-f` and/or `-r` to `rm` command when not needed
- Refactored the symlink naming logic to reduce code redundancy
- Added support for the workshop folders as a bonus :P

**Comment by Spiricore** [comment]
Thanks for updating it! 

Unfortunately, the multiple drives is not something I can test on my Steam Deck either.
I just switched from Win11 to CachyOS (Arch) on desktop with the multiple drives and was using the Desktop version of shortix from here: https://github.com/SeongGino/shortix

I'm unclear how to use this on desktop to test it there which is why I used the fork from SeongGino. Is there a way to do that with your release as well?

**Comment by Snogard** [comment]
i wrote the instructions on my previous comment XD.
Anyway here is a steby-by-step:
1. make sure you have python and protontricks installed
2. `git clone https://github.com/Snogard/shortix.git`
3. `cd shortix`
4. `git checkout improvements`
5. `bash ./simulate_installer.sh`

I'll check out the fork you mentioned to see if there's something worth adding.

**Comment by Spiricore** [comment]
Thank you for giving the step-by-step. That helped!

Your fork appears to be locating all of my drives. Huzzah!

The only immediate issue I see is that options for simulate_installer don't seem to apply to folder names.
...and having 'Shortix' and 'shortix' folders in the same dir.

**Comment by Snogard** [comment]
Can you upload here a screenshot of your:
- Folder where shortix and Shortix are present
- $HOME/.config/Shortix folder
- $HOME/Shortix
and i need you to tell me if you have the other fork you mentioned still installed.


Also, regarding the other fork, it appears to only makes changes to be able to install Shortix system wide. I'll integrate those and I'll make a AUR package too.

**Comment by Spiricore** [comment]
<img width="313" height="998" alt="Image" src="https://github.com/user-attachments/assets/9153b8e2-a173-494f-8cbf-d030978781b3" />
<img width="212" height="638" alt="Image" src="https://github.com/user-attachments/assets/f7135aea-0369-44dd-96e2-132451a29d12" />

I do not appear to have a shortix folder in .config (I can see hidden folders but have them listed below where I cut off the screenshot) and I uninstalled the desktop version before running your fork.
Thank you for your continued work!

**Comment by Snogard** [comment]
> The only immediate issue I see is that options for simulate_installer don't seem to apply to folder names. ...and having 'Shortix' and 'shortix' folders in the same dir.

The shortix and Shortix folders are there because you probably run my steby-by-step instructions directly in your home directory while in your terminal and the smaller case one is likely my repository that you cloned. That won't cause issues because linux paths are case sensitive.

> I do not appear to have a shortix folder in .config (I can see hidden folders but have them listed below where I cut off the screenshot

The screenshots are fine. The missing .config/Shortix folder is the issue here, that's why your configs didn't apply, there aren't any XD. I'll try to investigate why the config folder and files weren't generated.

**Comment by Snogard** [comment]
Should be fixed now, i forgot to add the code to create that folder, silly me.

To test now that you already cloned:
1. `cd $HOME/shortix
2. `git pull`
3. `bash ./simulate_installer.sh`

or simply delete the shortix folder and repeat the guide from a few days ago.

Then, you either delete the contents of `$HOME/.cache/Shortix` and execute shortix.sh or you can directly with `-f flag` `./shortix.sh -f` to force its execution.

If everything goes well i'll make the merge request and once merged i'll make an updated AUR package

**Comment by Spiricore** [comment]
Folder info is working now as well as the .cache folder being created.
And yeah, the shortix folder is being created by the git pull and the Shortix folder is being created by actually running it. 
The only "issue" I see left is the script is still trying to delete a .desktop shortcut used for Steam Deck users at the end. I assume it's just running as intended:  rm: cannot remove '/home/spiripc/Desktop/shortix_installer.desktop': No such file or directory
Seems good to push to the AUR. Fantastic work, both of you!

**Comment by Snogard** [comment]
> The only "issue" I see left is the script is still trying to delete a .desktop shortcut used for Steam Deck users at the end. I assume it's just running as intended: rm: cannot remove '/home/spiripc/Desktop/shortix_installer.desktop': No such file or directory

Not 100% sure but i think this is intended behaviour.
